smt/applications/egop.py

```diff
@@ -0,0 +1,240 @@
+"""
+Author: Emile Roux <emile.roux@univ-smb.fr>
+
+Drived from the EGO class by Remi Lafage <remi.lafage@onera.fr>, Nathalie Bartoli
+
+This package is distributed under New BSD license.
+
+
+"""
+
+
+from __future__ import division
+import six
+import numpy as np
+
+from types import FunctionType
+
+from scipy.stats import norm
+from scipy.optimize import minimize
+
+from smt.utils.options_dictionary import OptionsDictionary
+from smt.applications.ego import EGO
+
+from smt.utils.misc import compute_rms_error
+
+from smt.surrogate_models import KPLS, KRG, KPLSK
+from smt.sampling_methods import LHS
+
+
+
+
+class EGO_para(EGO):
+    """
+    The EGO_para class provide a prallel version of the EGO class.
+    The basic idea is to extract n intersting point at once and then evaluates 
+    this n points in parallel
+    """
+    def _initialize(self):
+        super(EGO_para, self)._initialize()
+        declare = self.options.declare
+        declare(
+            "criterion",
+            "EI",
+            types=str,
+            values=["EI", "SBO", "UCB"],
+            desc="criterion for next evaluation point determination: Expected Improvement, \
+            Surrogate-Based Optimization or Upper Confidence Bound",
+        )
+        declare("n_iter", None, types=int, desc="Number of optimizer steps")
+        declare(
+            "n_max_optim",
+            20,
+            types=int,
+            desc="Maximum number of internal optimizations",
+        )
+        declare("n_start", 20, types=int, desc="Number of optimization start points")
+        declare("n_par", 1, types=int, desc="Number parallel sample the compute using the qEI ")
+        declare("qEIAproxCrit", "KBLB", types=str,
+                values=["KB", "KBLB", "KBUB", "KBRand", "CLmin"],
+                desc="Approximated q-EI maximization strategy ")
+        declare(
+            "n_doe",
+            None,
+            types=int,
+            desc="Number of points of the initial LHS doe, only used if xdoe is not given",
+        )
+        declare("xdoe", None, types=np.ndarray, desc="Initial doe inputs")
+        declare("ydoe", None, types=np.ndarray, desc="Initial doe outputs")
+        declare("xlimits", None, types=np.ndarray, desc="Bounds of function fun inputs")
+        declare("verbose", False, types=bool, desc="Print computation information")
+
+    def optimize(self, fun, JobRunner=None, **kwargs):
```

Comment:
I noticed you have suppressed JobRunner class as it was not py27 compatible but we may re-introduce it as we are heading to a drop of py2.7 support and state that this feature works only in Python 3.